agent/consul/state/config_entry.go

```diff
@@ -237,7 +237,18 @@ func ensureConfigEntryTxn(tx WriteTxn, idx uint64, conf structs.ConfigEntry) err
 				return err
 			}
 		}
+
+		if !statusUpdate {
+			if controlledConf, ok := conf.(structs.ControlledConfigEntry); ok {
+				controlledConf.SetStatus(existing.(structs.ControlledConfigEntry).GetStatus())
+			}
+		}
 	} else {
+		if !statusUpdate {
+			if controlledConf, ok := conf.(structs.ControlledConfigEntry); ok {
+				controlledConf.SetStatus(controlledConf.DefaultStatus())
+			}
+		}
```

Comment:
Thinking out loud here, it seems that in **most** cases, `conf` is _not_ a `ControlledConfigEntry` whenever `statusUpdate == false`; however, when `conf` _is_ a `ControlledConfigEntry`, we want to reset the status to whatever already exists in committed state or the default status if no committed state. Accurate?

Reply:
Yep, so if it's a create operation for a `ControlledConfigEntry` and you're using one of the user-facing operations that shouldn't touch status, then set the status to `DefaultStatus`; if it's an update, then just ensure that the status is not changed by merging in the currently stored status.